ext/MatrixAlgebraKitAMDGPUExt/MatrixAlgebraKitAMDGPUExt.jl

```diff
@@ -128,17 +128,23 @@ function MatrixAlgebraKit._project_hermitian_diag!(A::StridedROCMatrix, B::Strid
 end
 
 MatrixAlgebraKit.ishermitian_exact(A::StridedROCMatrix) = all(A .== adjoint(A))
-MatrixAlgebraKit.ishermitian_exact(A::Diagonal{T, <:StridedROCVector{T}}) where {T} =
-    all(A.diag .== adjoint(A.diag))
+MatrixAlgebraKit.ishermitian_exact(A::Diagonal{T, <:StridedROCVector{T}}) where {T <: Real} = true
+function MatrixAlgebraKit.ishermitian_exact(A::Diagonal{T, <:StridedROCVector{T}}) where {T <: Complex}
+    return all(isreal.(A.diag))
+end
 MatrixAlgebraKit.ishermitian_approx(A::StridedROCMatrix; kwargs...) =
     @invoke MatrixAlgebraKit.ishermitian_approx(A::Any; kwargs...)
 
 MatrixAlgebraKit.isantihermitian_exact(A::StridedROCMatrix) =
     all(A .== -adjoint(A))
-MatrixAlgebraKit.isantihermitian_exact(A::Diagonal{T, <:StridedROCVector{T}}) where {T} =
-    all(A.diag .== -adjoint(A.diag))
+MatrixAlgebraKit.isantihermitian_exact(A::Diagonal{T, <:StridedROCVector{T}}) where {T <: Real} = all(iszero(A.diag))
+MatrixAlgebraKit.isantihermitian_exact(A::Diagonal{T, <:StridedROCVector{T}}) where {T <: Complex} =
+    all(iszero.(real.(A.diag)))
 MatrixAlgebraKit.isantihermitian_approx(A::StridedROCMatrix; kwargs...) =
     @invoke MatrixAlgebraKit.isantihermitian_approx(A::Any; kwargs...)
+function MatrixAlgebraKit.isantihermitian_approx(A::Diagonal{T, <:StridedROCVector{T}}; atol, rtol, kwargs...) where {T <: Real}
+    return sum(abs2, A.diag) ≤ max(atol, rtol * norm(A))
```

Comment:
This is a strange condition, as the left hand side is `norm(A)^2`, so this is saying `norm(A)^2 < max(atol, rtol * norm(A))`

This amounts to `norm(A)^2 < atol` or `norm(A) < rtol`, both of which look off.

Because a real diagonal matrix equals its hermitian project and has zero non-hermitian projection, I think the condition simply needs to be `norm(A) < atol` to be compatible with how approximate antithermiticity is checked elsewhere.